src/traces/densitymapbox/plot.js

```diff
@@ -0,0 +1,131 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var convert = require('./convert');
+
+function DensityMapbox(subplot, uid) {
+    this.subplot = subplot;
+    this.uid = uid;
+
+    this.sourceId = uid + '-source';
+
+    this.layerList = [
+        ['heatmap', uid + '-layer-heatmap']
+    ];
+
+    // previous 'below' value,
+    // need this to update it properly
+    this.below = null;
+}
+
+var proto = DensityMapbox.prototype;
+
+proto.update = function(calcTrace) {
```

Comment:
Also very similar to the function in `src/traces/choroplethmapbox/plot.js`. I'll leave it up to you to decide whether to reuse it or not.